pkg/cloudprovider/backup_service.go

```diff
@@ -161,7 +166,7 @@ func (br *backupService) GetAllBackups(bucket string) ([]*api.Backup, error) {
 	output := make([]*api.Backup, 0, len(prefixes))
 
 	for _, backupDir := range prefixes {
-		backup, err := br.GetBackup(bucket, backupDir)
+		backup, err := br.GetBackup(bucket, bucketPath, backupDir)
```

Comment:
the prefixes that come back from ListCommonPrefixes include the path. so when we call GetBackup here we end up double-prefixing with the path. Probably makes sense to strip the path off of `backupDir` before passing it to GetBackup